astrokat/simulate.py

```diff
@@ -147,74 +183,138 @@ def __exit__(self, type, value, traceback):
         # TODO: self.track_ cleanup for multiple obs loops
         if self.track_:
             self.kat._session_cnt += 1
-        if self.kat._session_cnt < len(self.obs_params['observation_loop']):
-            self.kat._lst, _ = get_lst(self.obs_params['observation_loop'][self.kat._session_cnt]['LST'])
+        if self.kat._session_cnt < len(self.obs_params["observation_loop"]):
+            self.kat._lst, _ = get_lst(
+                self.obs_params["observation_loop"][self.kat._session_cnt]["LST"]
+            )
 
     def _fake_slew_(self, target):
         slew_time = 0
         if not (target == self.katpt_current):
             if self.katpt_current is None:
                 slew_time = _DEFAULT_SLEW_TIME
             else:
-                user_logger.debug('Slewing to {}'.format(target.name))
+                user_logger.debug("Slewing to {}".format(target.name))
                 slew_time = self.slew_time(target)
         return slew_time
 
     def track(self, target, duration=0, announce=False):
+        """Simulate the track source functionality during observations.
+
+        Parameters
+        ----------
+        target: katpoint.Target object
```

Comment:
When should not include "object" in this, `katpoint.Target` is sufficient - it describes the type.